src/util/indent/lexi.c

```diff
@@ -372,6 +378,19 @@ lexi()
 	break;
 
     case '#':
+	if (lpc && *buf_ptr == '\'' ) { /* it's a closure */
+	    buf_ptr++;
+	    while(!(isspace(*buf_ptr)||*buf_ptr=='{'||*buf_ptr=='}'))
```

Comment:
The dereference of buf_ptr is done without checking that it's still within the buffer.

Reply:
My C is not good. I take it I need to test on buf_end somehow, would it be enough to reorder the body of the while loop to do if(buf_ptr >= buf_end) fill_buffer(); before the if(*buf_ptr == ',') or do I have a problem outside the while loop on line 382?